src/transformers/trainer.py

```diff
@@ -101,6 +101,11 @@
     _use_native_amp = True
     from torch.cuda.amp import autocast
 
+if version.parse(torch.__version__) < version.parse("1.2"):
+    _use_ddp_no_sync = False
+else:
+    _use_ddp_no_sync = True
+
```

Comment:
We only support pytorch > 1.3.1 for the examples, so you can remove this if/else clause!

Reply:
Great, so I removed this check

Reply:
This is not an example though, and `Trainer` is in the main lib. So we should keep this check until we pin pytorch to a later version IMO.

Reply:
@sgugger is right.

Reply:
I reverted to the original implementation

Reply:
Thank you!